src/gui/StelGuiItems.cpp

```diff
@@ -1030,29 +1030,36 @@ void BottomStelBar::updateText(bool updatePos, bool updateTopocentric)
 
 	if (updatePos)
 	{
+		// The location string starts at left, dateTime ends at right. FoV and FPS come ahead of dateTime.
+		// In case the strings are wider than available button space, they are now pushed to the right.
 		QFontMetrics fpsMetrics(fps->font());
-		int fpsShift = fpsMetrics.boundingRect(fpsstr).width() + 50;
+		int fpsShift = fpsMetrics.boundingRect(fpsstr).width() + fpsMetrics.boundingRect("MMMM").width();
 
 		QFontMetrics fovMetrics(fov->font());
-		int fovShift = fpsShift + fovMetrics.boundingRect(fovstr).width() + 80;
+		int fovShift = fpsShift + fovMetrics.boundingRect(fovstr).width() + fovMetrics.boundingRect("MMMM").width();
 		if (getFlagFovDms())
-			fovShift += 25;
+			fovShift += fovMetrics.boundingRect("MM'SS\"").width();
 
 		QRectF rectCh = getButtonsBoundingRect();
-		location->setPos(0, 0);		
-		int dtp = static_cast<int>(rectCh.right()-datetime->boundingRect().width())-5;
-		if ((dtp%2) == 1) dtp--; // make even pixel
-		datetime->setPos(dtp,0);
+		location->setPos(0, 0);
+		QFontMetrics locationMetrics(fov->font());
+		int locationWidth=locationMetrics.boundingRect(location->text()).width() + locationMetrics.boundingRect("MMM").width();
+
+		int dateTimePos = static_cast<int>(rectCh.right()-datetime->boundingRect().width())-5;
```

Comment:
Since you're replacing the numbers by semantically-loaded expressions, why is this -5 still here as a literal?

Reply:
I actually just *did not change* this buffer width of 5 pixels. Whoever introduced "5" likely did so for aesthetic reasons. It could probably be replaced by something like 'StelBarsPath::roundSize-1'. 
I just saw that CornerButtons::boundingRect() also returns a boundingRect that appears to be 1 pixel too small in my view. I don't understand why boundingRect sizes are reduced by 1. Insights welcome.

Reply:
> I actually just _did not change_

Yes, this is what makes me wonder, because you did change some magic numbers above but didn't change this one.

Reply:
Because I did not see an aesthetic issue about the change in X coordinates. You see the purpose of this, "keep a small distance from the right edge". This still works. Keeping the constant may be faster than retrieving the 'StelBarsPath::roundSize-1' which is actually still just a named constant without in-program setting option.